src/Microsoft.AspNetCore.Razor.Language/Intermediate/RazorIRNode.cs

```diff
@@ -8,15 +8,20 @@ namespace Microsoft.AspNetCore.Razor.Language.Intermediate
     public abstract class RazorIRNode
     {
         internal static readonly RazorIRNode[] EmptyArray = new RazorIRNode[0];
+        internal static readonly RazorDiagnostic[] EmptyDiagnostics = new RazorDiagnostic[0];
 
         public abstract ItemCollection Annotations { get; }
 
+        public abstract IList<RazorDiagnostic> Diagnostics { get; }
+
         public abstract IList<RazorIRNode> Children { get; }
 
         public abstract RazorIRNode Parent { get; set; }
 
         public abstract SourceSpan? Source { get; set; }
 
         public abstract void Accept(RazorIRNodeVisitor visitor);
+
+        public abstract bool ContainsDiagnostics();
```

Comment:
Suggest renaming to `HasErrors` like we have in all our TagHelperDescriptor apis